packages/pieces/bettermode/src/lib/actions/assign-badge.ts

```diff
@@ -0,0 +1,34 @@
+import { createAction, Property } from "@activepieces/pieces-framework";
+import { assignBadgeToMember } from "../api";
+import { buildBadgesDropdown } from "../props";
+import { bettermodeAuth, BettermodeAuthType } from "../auth";
+
+export const assignBadgeAction = createAction({
+	name        : 'assignBadge',
+	auth        : bettermodeAuth,
+	displayName : 'Assign Badge to Member',
+	description : 'Assign an existing badge to a member by email',
+	props       : {
+		badgeId: Property.Dropdown({
+			displayName : 'Badge',
+			description : 'The badge to assign',
+			required    : true,
+			refreshers  : ['auth'],
```

Comment:
You don't need to put auth in dropdown refreshers anymore, it should happen automatically

Reply:
@MoShizzle When I remove this, it generates TS errors for me. I have to keep refreshers in there with an empty array to not get TS errors. This may want to get fixed if it's not really needed.